packages/playwright-core/src/server/recorder/codeGenerator.ts

```diff
@@ -141,12 +141,11 @@ export class CodeGenerator extends EventEmitter {
       return;
     }
 
-    if (signal.name === 'navigation') {
+    if (signal.name === 'navigation' && frame._page.mainFrame() === frame) {
```

Comment:
This moved the check from the individual language generator one layer up to where it gets dispatched since navigate can only be issued on the main frame.

(Also to make recorderActions.ts typings work)